                        POISSON DISTRIBUTION

A poisson distribution is a discrete probability distribution that expresses the probability of a given number of events occurring in a fixed interval of time or space if these events occur with a known constant mean rate and independently of the time since the last event. The Poisson distribution can also be used for the number of events in other specified intervals such as distance, area or volume.

For instance:
+ A hospital records an average of 200 new born babies per hour (λ = babies/hour). 
+ A police station receives an average of 15 calls every three hours (λ = 15 calls/3hours)

Formal Definition: A discrete random variable X has a Poisson distribution with parameter λ if for k=0,1,2.., the probability mass function of X is given by:

$$Pr(X=k)=\frac{\lambda^ke^{-\lambda}}{k!}$$
where e is Euler's number (e=2.718...) and k! is the factorial of k.

  Assumptions and validity:
The Poisson distribution is an appropriate model if the following assumptions are true:
+ k is the number of times an event occurs in an interval and k can take values 0, 1, 2, ...
+ The occurrence of one event does not affect the probability that a second event will occur. That is, events occur independently.
+ The average rate at which events occur is independent of any occurrences. For simplicity, this is usually assumed to be constant, but may in practice vary with time.
+ Two events cannot occur at exactly the same instant; instead, at each very small sub-interval exactly one event either occurs or does not occur.

Let's see some exercises




Exercise 1: Suppose the average number of car accidents on the highway in one day is 4. 

a/ What is the probability of no car accident in one day?

b/ What is the probability of 1 car accident in two days?

In [8]:
# Ex 1a: First we have to set variables and import math library
lamb = 4 # λ = 4 car accidents/day
k = 0

# Import math library
from math import factorial
from math import exp

# Compute the probability of 0 car accident in one day
prob = (lamb**k)*exp(-lamb)/factorial(k)

# Print the output
print(prob)


0.01831563888873418


There is a 1.83 % chance that no car accident in one day

In [9]:
# Ex 1b: 
lamb2 = 8 # λ1 = 4 car accidents/day => λ2 = 8 car accidents/2days
k = 1

# Import math library
from math import factorial
from math import exp

# Compute the probability of 1 car accident in two days
prob = (lamb2**k)*exp(-lamb2)/factorial(k)

# Print the output
print(prob)


0.002683701023220095


There is a 0.268 % chance that 1 car accident in two days

Exercise 2: Suppose an average number of calls by 104 in one minute is 2. 

a/ What is the probability of 10 calls in 5 minutes?

b/ What is the probability of less than 10 calls in 5 minutes?

c/ What is the probability of at least 10 calls in 5 minutes?

d/ What is the probability of exactly 10 calls in 30 seconds?

In [12]:
# Ex 2a: Set variables
lamb = 10  # λ = 2*5 calls/5 minutes
k = 10

# Import math library
from math import factorial
from math import exp


# Compute the probability of 1 car accident in two days
prob = (lamb**k)*exp(-lamb)/factorial(k)

# Print the output
print(prob)



0.1251100357211333


There is a 12.511 % chance that the center receives 10 calls in 5 minutes

In [31]:
# Ex 2b:
# First, install python -m pip install --user numpy scipy matplotlib ipython jupyter pandas sympy nose
from scipy.stats import poisson

# Calculate probability of less 10 calls in 5 minutes
# k < 10 and λ = 2*5 calls/5 minutes
# Since k is an integer and less than 10 => k <= 9
poisson.cdf(k=9, mu =10)


0.4579297144718523

There is a 45.793 % chance that the center receives less than 10 calls in 5 minutes

In [36]:
# Ex 2c:
from scipy.stats import poisson

# k >= 10 and λ = 2*5 calls/5 minutes
# Calculate the probability of at least 10 calls in 5 minutes by 1 - p(k <= 9)
1 - poisson.cdf(k=9, mu =10)


0.5420702855281476

There is a 54.207 % chance that the center receives at least 10 calls in 5 minutes

Ex 4d:  What is the probability of exactly 10 calls in 30 seconds?

The initial λ = 2 calls/minute

Now we want to compute the probability of 10 calls in 30 seconds

Which means the new λ' = λ/2 (This is not accurate)

The new λ is deduced from the old one, thus, when we divide λ by a number which makes the result might not be correct

For instance, 2^2 = 4

However, when you say square root of 4 is 2 which might not be correct because it can be -2

Therefore, this given information is not accurate and we can not compute the probability of receiving 10 calls i 30 seconds

